src/include/sof/lib/pm_runtime.h

```diff
@@ -47,8 +47,26 @@ enum pm_runtime_context {
 struct pm_runtime_data {
 	spinlock_t lock;	/**< lock mechanism */
 	void *platform_data;	/**< platform specific data */
+#if CONFIG_POWER_STATE_COUNTERS
+	struct p_counters_data *p_counters; /**< debug power state counters */
+#endif
 };
 
+#if CONFIG_POWER_STATE_COUNTERS
+/** \brief Power states */
+enum dsp_p_state {
+	r0_p_state = 0,
+	r1_p_state,
+	r2_p_state
```

Comment:
Can you describe what these states mean for doxygen. I've no idea what state r0 is etc.

Reply:
Added small comment but note that they are abstract so it depends on platform implementation what state r0 means

+rebased to master